src/Compilers/Core/Portable/MetadataReader/MetadataTypeName.cs

```diff
@@ -204,14 +266,27 @@ public string UnmangledTypeName
             {
                 if (_unmangledTypeName == null)
                 {
-                    Debug.Assert(_inferredArity == -1);
```

Comment:
>Debug.Assert(_inferredArity == -1); [](http://example.com/codeflow?start=20&length=35)

It is not obvious why the assert is removed. Is it because ```UnmangledTypeNameMemory``` was accessed and calculated ```_inferredArity``` before we get here? In that case ```UnmangledTypeNameMemory``` shouldn't have the assert either.

Reply:
good point :)